### ISSUE TITLE
Queries with different answers

### ISSUE THREAD

**Issue by yahi012** [souffle-lang/souffle#651]
I am wondering if you can tell me why these two queries return different answers?
```
.decl t(x:number, y:number)

t(1,2).
t(2,3).

.decl q1(x:number)
.decl q2(x:number)
.decl m(x:number)

.output q1(IO=stdout)
.output q2(IO=stdout)

m(n) :- n = max(y) : {t(_,y)}.
q1(x) :- t(x,y), m(y).

q2(x) :- n = max(y) : {t(_,y)}, t(x,n).
```
I thought `q2` is just the "inlined" version of `q1`.

**Comment by mmcgr** [comment]
That looks to be because of a bug with the handling of aggregates in sub-clauses.

q2 should indeed be just the inlined version of q1. However, aggregates currently have issues in some cases (#550, #564, #556), and this seems to be one of them.

**Comment by mmcgr** [comment]
Closed in favour of #662.
